File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand1, strand2)
    arr1 = strand1.split('')
    arr2 = strand2.split('')

    count = 0
    arr1.zip(arr2).each do |arr1, arr2|
      count += 1 if arr1 != arr2
    end
    count
  end
end

File: javascript/hamming/hamming.js
exports.compute = function(strand1, strand2) {
  if (strand1.length !== strand2.length) throw new Error('DNA strands must be of equal length.');
  
  var count = 0;
  for (var i = 0, len = strand1.length; i < len; i++) {
    if (strand1[i] !== strand2[i]) count ++;
  };
  return count;
};
